### PR TITLE
ref: Adjust log statements for consistency; remove redundant logs

### PR DESCRIPTION
**Pull request by lgarber-akamai**
## 📝 Description

This pull request updates all resources and data sources to normalize all debug log statements, fill in logging gaps, and remove redundant log statements. Additionally, this PR makes a few tangentially related code improvements.

## ✔️ How to Test

The following test steps assume you have pulled down this PR locally.

### Integration Testing

```
TF_LOG=DEBUG make int-test
```

### CODE REVIEW

File: linode/domain/framework_datasource.go [hunk 1]
```diff
@@ -83,13 +83,17 @@ func (d *DataSource) getDomainByID(ctx context.Context, id int) (*linodego.Domai
...
-	filter, _ := json.Marshal(map[string]interface{}{"domain": domain})
-
-	tflog.Trace(ctx, "client.ListDomains(...)")
+	filter, err := json.Marshal(map[string]interface{}{"domain": domain})
+	if err != nil {
+		return nil, diag.NewErrorDiagnostic(
+			"Failed to marshal domain filter",
+			err.Error(),
+		)
+	}
```
Comment (lgarber-akamai) [inline linode/domain/framework_datasource.go:96]: Looks like there was some missing error handling here 👀

**Comment by lgarber-akamai** [comment]
E2E test workflow: https://github.com/linode/terraform-provider-linode/actions/runs/8541372187

**Review by jriddle-linode — APPROVED** [review]
LGTM. Thankless job

**Review by zliang-akamai — APPROVED** [review]
Thanks for the cleaning up!

**Pull request merged**